edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Agent.cs

```diff
@@ -202,11 +209,34 @@ public async Task ReconcileAsync(CancellationToken token)
                     }
                 }
 
-                await this.reporter.ReportAsync(token, moduleSetToReport, await this.environment.GetRuntimeInfoAsync(), this.currentConfig.Version, this.status);
+                if (await this.IsServiceAvailableViaTcpAsync())
+                {
+                    await this.reporter.ReportAsync(token, moduleSetToReport, await this.environment.GetRuntimeInfoAsync(), this.currentConfig.Version, this.status);
```

Comment:
What happens if we don't call ReportAsync because the service isn't available? Will any information be lost (whatever was supposed to be reported)? Or will it just get reported later when we're back online?